builder/src/merge.rs

```diff
@@ -83,49 +83,56 @@ impl Merger {
         target: ArtifactStorage,
         chunk_dict: Option<PathBuf>,
         config_v2: Arc<ConfigV2>,
-    ) -> Result<BuildOutput> {
+    ) -> Result<BuildOutput, MergeError> {
         if sources.is_empty() {
-            bail!("source bootstrap list is empty , at least one bootstrap is required");
+            Err(MergeError::Other(anyhow!(
+                "source bootstrap list is empty , at least one bootstrap is required"
+            )))?
         }
         if let Some(digests) = blob_digests.as_ref() {
-            ensure!(
-                digests.len() == sources.len(),
-                "number of blob digest entries {} doesn't match number of sources {}",
-                digests.len(),
-                sources.len(),
-            );
+            if digests.len() != sources.len() {
+                Err(MergeError::Other(anyhow!(
```

Comment:
Does this let the `merge` func return the error immediately?

Reply:
maybe we can do it use `return`

Reply:
Maybe we can use `return`.